test/cmake/seqan3_require_benchmark.cmake

```diff
@@ -0,0 +1,47 @@
+# -----------------------------------------------------------------------------------------------------
+# Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+# Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+# This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+# shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+# -----------------------------------------------------------------------------------------------------
+
+# Exposes the google-benchmark target `gbenchmark`.
+macro (seqan3_require_benchmark)
+    enable_testing ()
+
+    set (gbenchmark_project_args ${SEQAN3_EXTERNAL_PROJECT_CMAKE_ARGS})
+    list (APPEND gbenchmark_project_args "-DBENCHMARK_ENABLE_TESTING=false")
+    # list (APPEND gbenchmark_project_args "-DBENCHMARK_ENABLE_LTO=true")
```

Comment:
```suggestion
    # google-benchmarks suggest to use LTO (link-time optimisation), but we don't really need that because we are a
    # header only library. This option might be still interesting for external libraries benchmarks.
    # see https://github.com/google/benchmark#debug-vs-release
    # list (APPEND gbenchmark_project_args "-DBENCHMARK_ENABLE_LTO=true")
```